games-emulation/rmg/rmg-0.6.5.ebuild

```diff
@@ -0,0 +1,134 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+CARGO_OPTIONAL=1
+
+CRATES="
+	cc@1.0.83
+	cfg-if@1.0.0
+	libc@0.2.152
+	libloading@0.7.4
+	libusb1-sys@0.6.4
+	once_cell@1.19.0
+	pkg-config@0.3.29
+	proc-macro2@1.0.78
+	quote@1.0.35
+	rusb@0.9.3
+	serde@1.0.195
+	serde_derive@1.0.195
+	syn@2.0.48
+	toml@0.5.11
+	unicode-ident@1.0.12
+	vcpkg@0.2.15
+	winapi-i686-pc-windows-gnu@0.4.0
+	winapi-x86_64-pc-windows-gnu@0.4.0
+	winapi@0.3.9
+"
+
+inherit cargo cmake xdg
+
+MY_PN="${PN^^}"
+MY_P="${MY_PN}-${PV}"
+DESCRIPTION="Rosalie's Mupen GUI"
+HOMEPAGE="https://github.com/Rosalie241/RMG"
+if [[ "${PV}" == *9999 ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com//Rosalie241/${MY_PN}.git"
+else
+	SRC_URI="
+		https://github.com//Rosalie241/${MY_PN}/archive/v${PV}/${MY_P}.tar.gz \
+			-> ${P}.tar.gz
+		rust? ( ${CARGO_CRATE_URIS} )
+	"
+	S="${WORKDIR}/${MY_P}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="
+	BSD-2 CC0-1.0 GPL-2 GPL-3 MIT ZLIB public-domain
+	angrylion? ( XMAME )
+	rust? ( ISC Unicode-DFS-2016 )
+"
+SLOT="0"
+IUSE="angrylion asm discord rust"
+REQUIRED_USE="asm? ( || ( amd64 arm arm64 x86 ) )"
```

Comment:
While I don't see anything exceptionally wrong with this, we prefer to use profile (use) masking for these. I'd mask the use flag on base profile, then unmask it on supported arches (amd64). Then again, do you expect to keyword this on other arches at all?

Reply:
Thanks for explaining this, I wasn't sure what the best method was? I added the profile masking in a new commit, I believe I did this correctly?

Reply:
Yep, looks correct!